connect/runtime/src/test/java/org/apache/kafka/connect/integration/TaskHandle.java

```diff
@@ -135,7 +136,7 @@ public int partitionsAssigned() {
      * @param  timeout duration to wait for records
      * @throws InterruptedException if another threads interrupts this one while waiting for records
      */
-    public void awaitRecords(int timeout) throws InterruptedException {
+    public void awaitRecords(long timeout) throws InterruptedException {
```

Comment:
could we add a unit like `timeoutMs`?

Reply:
I was trying to avoid lots of changes, but I agree it's useful. We do want to keep these test APIs backward compatible because the integration test framework is used outside of AK, so I'll add overloaded methods that take the unit.

Reply:
Actually I have the opposite comment to make. 
I wonder how useful it is to make this very generic and add a lot of boilerplate. In practice we'll never use in tests anything other than milliseconds. To the contrary we should discourage anyone doing so, because this is a real-time (not mocked) delay and delays in the order of multiple seconds are problematic. Even if we needed a few minutes (which I hope we'll never will), `TimeUnit.MILLISECONDS` is more than enough.

Reply:
I guess I disagree. I don't think we need minutes, but we may need 10s of seconds, and it's too easy to mistype or misread the number of zeros in `10000`. I would much rather use `awaitRecords(10, TimeUnit.SECONDS)`. And this is the convention used in all of the JDK for awaiting for something to happen, and I think sticking with JDK conventions is good.

Reply:
I don't feel too strong about this change. My points where a) to discourage using seconds for delays b) that this is not a general library such as the JDK or other in which I'd be all up for the fully customizable method version. Seems fine to leave as is though and not block this PR on this suggestion.